content/app/models/file-path.js

```diff
@@ -0,0 +1,111 @@
+define("content/app/models/file-path", function(require, exports, module) {
+
+	/**
+	 * Simple class used for handling of file paths.
+	 */
+	var FilePath = Class({
+
+		$statics : {
+			REGEX_FOLDER_SEPARATOR : /\/|\)/g,
+			EXTENSION_SEPARATOR : '.',
+		},
+
+		/**
+		 * Class constructor
+		 * @param  	{String}	path 	Full file path.
+		 */
+		constructor : function(path) {
+			this._path = path;
+			this._name = null;
+			this._extension = null;
+			this._segments = null;
+			this._initialize();
+		},
+
+		/**
+		 * Name of the file without extension.
+		 * @type {String}
+		 */
+		name : {
+			get : function() {
+				return this._name;
+			}
+		},
+
+		/**
+		 * File extension (without leading '.');
+		 * @type {String}
+		 */
+		extension : {
+			get : function() {
+				return this._extension;
+			}
+		},
+
+		/**
+		 * File name with extension (example file.txt).
+		 * @type {String}
+		 */
+		nameWithExt : {
+			get : function() {
+				return this._name + FilePath.EXTENSION_SEPARATOR + this._extension;
+			}
+		},
+
+		/**
+		 * Full file path (example: a/b/c.txt)
+		 * @type {String}
+		 */
+		path : {
+			get : function() {
+				return this._path;
+			}
+		},
+
+		/**
+		 * Segments from which path is made. For path a/b/c/text.txt
+		 * result will be [a, b, c, text.txt].
+		 * @type {String[]}
+		 */
+		segments : {
+			get : function() {
+				return this._segments;
+			}
+		},
+
+		/**
+		 * Returns name of the first directory in path.
+		 * @type {String}
+		 */
+		dirname : {
```

Comment:
This won't be changed during Path instance lifetime, so it can be stored locally too. Right?